lib/webvalve/service_url_converter.rb

```diff
@@ -0,0 +1,20 @@
+module WebValve
+  class ServiceUrlConverter
+    TOKEN_BOUNDARY_CHARS = Regexp.escape('.:/?#@&=').freeze
+    WILDCARD_SUBSTITUTION = ('[^' + TOKEN_BOUNDARY_CHARS + ']*').freeze
+    URL_PREFIX_BOUNDARY = ('[' + TOKEN_BOUNDARY_CHARS + ']').freeze
+    URL_SUFFIX_PATTERN = ('(' + URL_PREFIX_BOUNDARY + '.*)?\z').freeze
```

Comment:
so, i'm trying to work this out. does this preserve the behavior of leaving the end of the url open-ended? 

right now we rely on this `WebMock` `#to_rack` mapping to set up the fake service url binding to the fake service instance.

```
WebMock.stub_request(
  :any,
  url_to_regexp(config.service_url)
).to_rack(FakeServiceWrapper.new(config))
```

in order for that to do the right thing, we need the regexp to match any path defined underneath the `service_url` prefix. e.g. `http://bank.com` as a `service_url` needs to match `http://bank.com` and `http://bank.com/*`. technically, the current implementation is more permissive in that it looks like it will allow `http://bank.commmmmm` which is not really desirable, but i hadn't realized it was possible until just now.

Reply:
it _looks_ like this is going to give us `\Ahttp://bank.com([.:/?#@&=].*)?\z` which seems like it will more appropriately cover the desired/intended space of url matches now, which is great. though it might be a "breaking change" if we're being pedantic. admittedly, it's a good breaking change though.

Reply:
Yeah, if you look at the test suite I think you'll see that it allows arbitrary suffixes. The way it works is "if there's more random stuff (i.e. `.*`), there has to be a boundary first. One thing I'm realizing I'm not 100% sure of is whether `http://bank.com/` will be OK which I gotta check right now.

Reply:
Indeed it was broken, and to fix I had to make the suffix twice as ugly :)